src/Regex.cpp

```diff
@@ -35,7 +52,9 @@ FindRegexMatch::FindRegexMatch(const boost::regex &regexImpl, const std::string
             ++i;
         for (; i != matchResults.end(); ++i) {
             if (i->matched) {
-                RegexSubmatch s = {*i, i->first - expression.begin()};
+                std::ptrdiff_t codeUnitPosition = i->first - expression.begin();
+                RegexSubmatch s
+                    = {*i, utf8CodepointOffset(expression, static_cast<size_t>(codeUnitPosition))};
```

Comment:
What about a `RegexSubmatch s = {*i, utf8CodepointOffset(expression, i->first)};` and remove `codeUnitPosition` altogether?

By changing the signature of `utf8CodepointOffset`, we wouldn't even need to go from pointer (`i->first`) to diff (`codeUnitPosition`) to pointer again (`expression.begin() + codeUnitPosition`).

Reply:
@paoloambrosio I've addressed this, but left a `static_cast<size_t>` in the check for an out-of-range match.

Reply:
Hi @paoloambrosio,

I believe my first change was based on a misunderstanding of your suggestion. I've updated it once more. Let me know if this is what you were thinking.